### DISCUSSION TITLE
Problem with io redirection

### DISCUSSION THREAD

**Discussion by laguvajra** [official-stockfish/Stockfish#4624] [Q&A] [answered]
I am running stockfish 15.1 under windows 10. When I open a command window and run stockfish with out io redirection and just enter commands it seems to work fine.

E:\Lisp\Work>/lisp/system/chess/stockfish.exe
Stockfish 15.1 by the Stockfish developers (see AUTHORS file)
_position fen rnbqkb1r/pp2pppp/3p1n2/2p5/4P3/2P2N2/PPQP1PPP/RNB1KB1R b - -
isready
go depth 30_
info string NNUE evaluation using nn-ad9b42354671.nnue enabled
info depth 1 seldepth 1 multipv 1 score cp -7 nodes 35 nps 17500 hashfull 0 tbhits 0 time 2 pv e7e5
info depth 2 seldepth 2 multipv 1 score cp 15 nodes 97 nps 32333 hashfull 0 tbhits 0 time 3 pv b8c6
info depth 3 seldepth 2 multipv 1 score cp 15 nodes 214 nps 71333 hashfull 0 tbhits 0 time 3 pv b8c6
info depth 4 seldepth 2 multipv 1 score cp 15 nodes 247 nps 61750 hashfull 0 tbhits 0 time 4 pv b8c6
info depth 5 seldepth 2 multipv 1 score cp 21 nodes 289 nps 72250 hashfull 0 tbhits 0 time 4 pv b8c6
........ A whole bunch of more info stuff finally ending in .......
bestmove d8c7 ponder d2d4

My input are the three lines beginning with position, isready and go
When I run stockfish with input redirected, the exact same three lines, but coming from a file I get different results:

E:\Lisp\Work>/lisp/system/chess/stockfish.exe <engine_input.txt
Stockfish 15.1 by the Stockfish developers (see AUTHORS file)
readyok
info string NNUE evaluation using nn-ad9b42354671.nnue enabled
bestmove c5c4

With input redirection I no longer get all the info lines, and the final bestmove is different and also does not come with a ponder

Can anybody explain what is going on or what I am doing wrong?
Thanks

**Comment by vondele** [comment, marked as answer]
as you redirect the input, that input file is closed by the OS as soon as all input is sent. Once the input is closed SF will stop and quit the search. 

I would recommend to use python-chess for interaction with an UCI engine, but just keeping the input open until you see bestmove in the output works as well. 

The easiest might be to send a `ucinewgame` after the go command, which will wait until the search finishes.

**Comment by laguvajra** [comment]
Thank You, ucinewgame worked great!
Is there a way to disable all the info statements returned and get directly to the bestmove?

**Reply by Disservin** [reply]
There is not
